gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DistributedPutAllOperation.java

```diff
@@ -1339,7 +1340,8 @@ protected void basicOperateOnRegion(final EntryEventImpl ev, final DistributedRe
         rgn.syncPutAll(tx, new Runnable() {
           public void run() {
             UMMMemoryTracker memoryTracker = null;
-            if (CallbackFactoryProvider.getStoreCallbacks().isSnappyStore()) {
+            if (CallbackFactoryProvider.getStoreCallbacks().isSnappyStore()
+                    && !rgn.getName().toUpperCase().endsWith(StoreCallbacks.SHADOW_TABLE_SUFFIX)) {
```

Comment:
Can you add a utility method in LocalRegion like "isColumnStore()" and move this check there? It can be changed to be done in the constructor of LocalRegion itself for efficiency. Check many other cases of similar checks against SHADOW_TABLE_SUFFIX and fix all of those to use the method instead.